docs/configuration_options.rst

```diff
@@ -1,100 +1,110 @@
 Configuration Options
 ---------------------
 
-``connection.url``
-  The URL to connect to Elasticsearch.
+Connector
+^^^^^^^^^
 
-  * Type: string
-  * Default: ""
-  * Importance: high
-
-``type.name``
-  The type to use for each index.
+``connection.url``
+  Elasticsearch HTTP connection URL e.g. ``http://eshost:9200``.
 
   * Type: string
-  * Default: ""
-  * Importance: high
-
-``key.ignore``
-  Whether to ignore the key during indexing. When this is set to true, only the value from the message will be written to Elasticsearch. Note that this is a global config that applies to all topics. If this is set to true, Use ``topic.key.ignore`` to config for different topics. This value will be overridden by the per topic configuration.
-
-  * Type: boolean
-  * Default: false
   * Importance: high
 
 ``batch.size``
-  The number of requests to process as a batch when writing to Elasticsearch.
+  The number of records to process as a batch when writing to Elasticsearch.
 
   * Type: int
   * Default: 2000
   * Importance: medium
 
-``max.in.flight.requests``
-  The maximum number of incomplete batches each task will send before blocking.
+``max.buffered.records``
+  The maximum number of records each task will buffer before blocking acceptance of more records. This config can be used to limit the memory usage for each task.
 
   * Type: int
-  * Default: 5
-  * Importance: medium
+  * Default: 20000
+  * Importance: low
 
-``flush.timeout.ms``
-  The timeout when flushing data to Elasticsearch.
+``linger.ms``
+  Linger time in milliseconds for batching.
+
+  Records that arrive in between request transmissions are batched into a single bulk indexing request, based on the ``batch.size`` configuration. Normally this only occurs under load when records arrive faster than they can be sent out. However it may be desirable to reduce the number of requests even under light load and benefit from bulk indexing. This setting helps accomplishes that - rather than immediately sending out a record, the task will wait upto the given delay to allow other records to be added so that they can be batched into a single request.
 
   * Type: long
-  * Default: 10000
+  * Default: 1
   * Importance: low
 
-``linger.ms``
-  The task groups together any records that arrive in between request transmissions into a single batched request. Normally this occurs only under load when records arrive faster than they can be sent out. However in some circumstances the tasks may want to reduce the number of requests even under moderate load. This setting accomplishes this by adding a small amount of artificial delay. Rather than immediately sending out a record the task will wait for up to the given delay to allow other records to be sent so that the sends can be batched together.
+``flush.timeout.ms``
+  The timeout in milliseconds when flushing records to Elasticsearch.
```

Comment:
Seems worth documenting to me, will add a note (the task will fail, no retries).